lms/djangoapps/lti_provider/users.py

```diff
@@ -40,13 +53,17 @@ def authenticate_lti_user(request, lti_user_id, lti_consumer):
         if lti_consumer.require_user_account:
             # Verify that the email from the LTI Launch and the logged-in user are the same
             # before linking the LtiUser with the edx_user.
-            if request.user.is_authenticated and request.user.email.lower() == lis_email.lower():
-                lti_user = create_lti_user(lti_user_id, lti_consumer, request.user.email)
+            if request.user.is_authenticated and request.user.email.lower() == profile["email"]:
+                lti_user = create_lti_user(lti_user_id, lti_consumer, profile)
             else:
                 # Ask the user to login before linking.
                 raise PermissionDenied() from exc
         else:
-            lti_user = create_lti_user(lti_user_id, lti_consumer)
+            if lti_consumer.use_lti_pii:
+                profile["username"] = lti_user_id
+            else:
+                profile = None
+            lti_user = create_lti_user(lti_user_id, lti_consumer, profile)
```

Comment:
We can avoid putting all of it in the else block, right? Rather, put an explicit check with else if.
```suggestion
           elif lti_consumer.use_lti_pii:
                profile["username"] = lti_user_id
                lti_user = create_lti_user(lti_user_id, lti_consumer, profile)
            else:
                lti_user = create_lti_user(lti_user_id, lti_consumer, None)
```